server/github_test.go

```diff
@@ -283,6 +283,7 @@ func TestRateLimitTransport(t *testing.T) {
 		metricsMock.EXPECT().ObserveGithubRequestDuration(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
 		metricsMock.EXPECT().IncreaseGithubCacheMisses(gomock.Any(), gomock.Any()).AnyTimes()
 		metricsMock.EXPECT().IncreaseGithubCacheHits(gomock.Any(), gomock.Any()).AnyTimes()
+		metricsMock.EXPECT().IncreaseRateLimiterErrors().AnyTimes()
```

Comment:
It should be possible to add a test that returns a 403 with the correct error message and confirm that the metric is being incremented exactly once to have a stricter check.